src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/Diff/LoadoutDiskSateDiffUtils.cs

```diff
@@ -0,0 +1,124 @@
+using System.Diagnostics;
+using NexusMods.Abstractions.DiskState;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Games.DTO;
+using NexusMods.Abstractions.Games.Trees;
+using NexusMods.Abstractions.Loadouts.Files;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+/// <summary>
+/// Provides utility methods for diffing DiskStates and loadouts
+/// </summary>
+public static class LoadoutSynchronizerDiffExtensions
+{
+    /// <summary>
+    /// Computes the difference between a loadout and a disk state, assuming the loadout to be the newer state.
+    /// </summary>
+    /// <param name="loadout">Newer state, e.g. unapplied loadout</param>
+    /// <param name="diskState">The old state, e.g. last applied DiskState</param>
+    /// <returns>A tree of all the files with associated <see cref="FileChangeType"/></returns>
+    public static async ValueTask<FileDiffTree> LoadoutToDiskDiff(
+        this IStandardizedLoadoutSynchronizer loadoutSynchronizer,
+        Loadout loadout,
+        DiskStateTree diskState)
+    {
+        var flattenedLoadout = await loadoutSynchronizer.LoadoutToFlattenedLoadout(loadout);
+        return await FlattenedLoadoutToDiskDiff(flattenedLoadout, diskState);
+    }
+
+    private static ValueTask<FileDiffTree> FlattenedLoadoutToDiskDiff(FlattenedLoadout flattenedLoadout, DiskStateTree diskState)
+    {
+        var loadoutFiles = flattenedLoadout.GetAllDescendentFiles();
+        var diskStateEntries = diskState.GetAllDescendentFiles();
+
+        Dictionary<GamePath, DiskDiffEntry> resultingItems = new();
```

Comment:
Can you set the initial capacity here?
`Max(loadoutFiles, diskStateEntries)` should be ok, that accounts for deletion.

Reply:
Deletions are included in the Dictionary, so Max wouldn't be accurate if the loadout has both deletions and additions.
I guess it would still be better than nothing.